fiftyone/core/models.py

```diff
@@ -1958,6 +1958,48 @@ def embed_all(self, args):
         return np.stack([self.embed(arg) for arg in args])
 
 
+class PromptMixin(object):
+    """Mixin for :class:`Model` classes that can generate prompt embeddings.
+
+    This mixin allows for the possibility that only some instances of a class
+    are capable of generating prompt embeddings, per the value of the
+    :meth:`can_embed_prompts` property.
+    """
+
+    @property
+    def can_embed_prompts(self):
+        """Whether this instance can generate prompt embeddings."""
+        raise NotImplementedError(
+            "subclasses must implement can_embed_prompts"
+        )
+
+    def embed_prompt(self, arg):
```

Comment:
similarly, since we aren't actually embedding anything, can we use something along the lines of `vectorize_prompt`, `encode_prompt`,  or `get_text_embeddings`